### PR TITLE
Create login, signup components

### PR DESCRIPTION
**Pull request by andrelandgraf**
Issue: #26

## Summary (1-2 sentences)

Follow-up PR to #27 to enhance login, signup layout and create re-usable components

## Details (reason and description of the changes)

- Added `Card` component
- Added `Input` component
- Moved existing components into new subfolders:
  - `marketing` for all landing page/show-off components
  - `ui` for all reusable baseline components
  - `layout` for all context-aware layout components of our root layout
- Enhanced `root.tsx` layout styling
- Updated `/login` and `/signup` styling

### Challenges (what problems did I face?) (optional)

Creating a scalable/maintainable root layout that is reusable across all pages is hard. For instance, we want the page footer to always be at the bottom of the page and the main content of the page to expand as much as needed. But we also want to allow each main content to define its background color. These kinds of restrictions require that the root/base layout is scattered across different routes/components, which makes it tricky to maintain.

### Alternative implementations (what alternatives were considered and why were they disregarded?) (optional)

Not aware of a better solution for now.

### Solution (how did I fix the problems?)

We defined a root flex layout in `root.tsx` which can be extended in nested pathless layout routes (like `_auth.tsx`). Nested layout routes are currently context-aware that the `root.tsx` layout is flex-based and act as flex children.

### Login

Screenshot 2023-07-31 at 9 02 44 PM

Screenshot 2023-07-31 at 9 02 30 PM

### Signup

Screenshot 2023-07-31 at 9 03 11 PM

Screenshot 2023-07-31 at 9 00 25 PM

Note: The purple icon is injected by my password manager and not part of the UI.

### CODE REVIEW

**Review by JacobBaqleh1 — APPROVED** [review]

**Comment by JacobBaqleh1** [comment]
Great work

File: app/components/ui/forms.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import type { InputHTMLAttributes, ReactNode } from 'react';
+
+type InputProps = InputHTMLAttributes<HTMLInputElement> & {
+  label: ReactNode;
+};
+
+export function Input({ label, ...props }: InputProps) {
+  return (
+    <label className="flex w-full">
+      <span className="sr-only">{label}</span>
```
Comment (shami-dev) [inline app/components/ui/forms.tsx:10]: That is a really great idea to make the Input field reusable and put it inside the label which is for screen readers!

File: app/routes/_auth.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Outlet } from '@remix-run/react';
+
+export default function Component() {
+  return (
+    <div className="grow py-20 lg:py-40 bg-secondary w-full flex flex-col items-center justify-center">
+      <Outlet />
+    </div>
+  );
+}
```
Comment (shami-dev) [inline app/routes/_auth.tsx:9]: A Pathless Layout Route from Remix looks awesome! They don't add any segments to the URL.:rocket:

**Review by shami-dev — APPROVED** [review]
Great work! Loved all new folder names.

**Pull request merged**